operator-common/src/main/java/io/strimzi/operator/common/Operator.java

```diff
@@ -94,9 +99,60 @@ default void reconcileThese(String trigger, Set<NamespaceAndName> desiredNames,
         return Optional.empty();
     }
 
-    Counter getPeriodicReconciliationsCounter();
+    Counter getPeriodicReconciliationsCounter(String namespace);
 
-    AtomicInteger getResourceCounter();
+    AtomicInteger getResourceCounter(String namespace);
 
-    AtomicInteger getPausedResourceCounter();
+    AtomicInteger getPausedResourceCounter(String namespace);
+
+    static Counter getCounter(String namespace, String kind, MetricsProvider metrics, Labels selectorLabels, Map<String, Counter> counterMap, String metricName, String metricHelp) {
+        String selectorValue = selectorLabels != null ? selectorLabels.toSelectorString() : "";
+        Tags metricTags = null;
+        if (namespace.equals("*")) {
+            metricTags = Tags.of(Tag.of("kind", kind), Tag.of("selector", selectorValue));
+        } else {
+            metricTags = Tags.of(Tag.of("kind", kind), Tag.of("namespace", namespace), Tag.of("selector", selectorValue));
+        }
+
+        Counter counter = counterMap.get(namespace + "/" + kind);
+        if (counter == null) {
+            counter = metrics.counter(metricName, metricHelp, metricTags);
+            counterMap.put(namespace + "/" + kind, counter);
+        }
+        return counter;
+    }
+
+    static AtomicInteger getGauge(String namespace, String kind, MetricsProvider metrics, Labels selectorLabels, Map<String, AtomicInteger> gaugeMap, String metricName, String metricHelp) {
+        String selectorValue = selectorLabels != null ? selectorLabels.toSelectorString() : "";
+        Tags metricTags = null;
+        if (namespace.equals("*")) {
+            metricTags = Tags.of(Tag.of("kind", kind), Tag.of("selector", selectorValue));
+        } else {
+            metricTags = Tags.of(Tag.of("kind", kind), Tag.of("namespace", namespace), Tag.of("selector", selectorValue));
+        }
+
+        AtomicInteger gauge = gaugeMap.get(namespace + "/" + kind);
+        if (gauge == null) {
+            gauge = metrics.gauge(metricName, metricHelp, metricTags);
+            gaugeMap.put(namespace + "/" + kind, gauge);
+        }
+        return gauge;
```

Comment:
This is basically identical to `getCounter` and `getTimer`, the only difference is which method of `metrics` we call. You should be able to write a single `private` generic method with abstracts across this. 

Something like 

```java
static <M> AtomicInteger metric(String namespace, String kind, MetricsProvider metrics, Labels selectorLabels, Map<String, AtomicInteger> gaugeMap, Function<Tags, M> fn) {
    //...
    return metrics.computeIfAbsent(metricName, x -> fn.apply(metricTags);
}
```

and implement each of `getGuage`, `getCounter` and `getTimer` with `return metric(..., tags -> metrics.guage(metricName, metricHelp, tags)`.